core/src/main/java/org/apache/calcite/rex/RexExecutorImpl.java

```diff
@@ -129,22 +129,20 @@ public static RexExecutable getExecutable(RexBuilder rexBuilder, List<RexNode> e
    */
   @Override public void reduce(RexBuilder rexBuilder, List<RexNode> constExps,
       List<RexNode> reducedValues) {
-    String code;
     try {
-      code = compile(rexBuilder, constExps, (list, index, storageType) -> {
+      String code = compile(rexBuilder, constExps, (list, index, storageType) -> {
         throw new UnsupportedOperationException();
       });
+
+      final RexExecutable executable = new RexExecutable(code, constExps);
+      executable.setDataContext(dataContext);
+      executable.reduce(rexBuilder, constExps, reducedValues);
```

Comment:
Is this change really needed? `RexExecutable#reduce` [has already its try-catch](https://github.com/apache/calcite/blob/60e0a3f441a009e55a36cac192253a436bec3f6d/core/src/main/java/org/apache/calcite/rex/RexExecutable.java#L81) and does `reducedValues.addAll(constExps);` in case of a problem

Reply:
Yes, let me attach a stack trace with an exception.

Reply:
The crash happens in the constructor for RexExecutable, so the try-catch block in `reduce` is too late.

Reply:
Ok, understood.